Java/measureeval/src/main/java/com/lantanagroup/link/measureeval/utils/CqlLogAppender.java

```diff
@@ -0,0 +1,93 @@
+package com.lantanagroup.link.measureeval.utils;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.AppenderBase;
+import com.lantanagroup.link.measureeval.services.MeasureEvaluator;
+import com.lantanagroup.link.measureeval.services.MeasureEvaluatorCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CqlLogAppender extends AppenderBase<ILoggingEvent> {
```

Comment:
Not critical, but since this needs the Spring application context, why not make it an `ApplicationContextAware` `@Component`?  Rather than statically calling into the new `ApplicationContextProvider` (which could then be removed).  Although with this strategy we'd need to add the appender to the Logback context programmatically, rather than specifying it in the XML config.  This is basically the "simple approach" described here:

https://stackoverflow.com/a/55575749/24515610

And on second thought, we _don't_ need the whole application context ... we just need a single bean.  Which we could wire in directly (as a constructor argument) if this class were a `@Component`.

Reply:
I was under the impression that `AppenderBase` is not instantiated using Spring, so attributing it with `@Component` wouldn't really do anything. I see this in the link you provided:

> It isn't possible to do what you are trying to do. Logback is initialised before the application context is created so there's nothing to perform the dependency injection.

And when I test it with `@Component` I get the following exception:

```
09:36:03,547 |-ERROR in ch.qos.logback.core.model.processor.AppenderModelHandler - Could not create an Appender of type [com.lantanagroup.link.measureeval.utils.CqlLogAppender]. ch.qos.logback.core.util.DynamicClassLoadingException: Failed to instantiate type com.lantanagroup.link.measureeval.utils.CqlLogAppender
	at ch.qos.logback.core.util.DynamicClassLoadingException: Failed to instantiate type com.lantanagroup.link.measureeval.utils.CqlLogAppender
```

Reply:
Hmm, tried to test this out myself, and I can't get _any_ CQL debugging output from the CLI.  Which I guess makes sense ... the `CqlLogAppender` ultimately relies on the Spring application context to get the measure definition bundle, but the CLI isn't a Spring application, so there _is_ no application context.  (This results in `append` throwing an NPE, which Logback swallows.)  So maybe a more important issue (and bigger refactor) than I originally thought.

Reply:
OK to resolve this conversation, then, and follow up more on it later?

Reply:
Fine by me if we're OK without CQL logging in the CLI.  But if that's the case, we should probably remove `CqlLogAppender` from the CLI's Logback config.

Reply:
Well... No. Not OK without CQL logging in the CLI. But, CQL logging in CLI works for me, using the currently implemented use of `ApplicationContextProvider`. If I replace that with something that uses @Component, _then_ I don't get any CQL logging in the CLI.

Reply:
Hmm, I must be running the CLI differently.  Are you running as described in the README (`mvn package` followed by `java -jar`)?  When I do that, I don't see any CQL logging.  And when I step through in IntelliJ using a run/debug configuration that references logback-cli.xml as the `logback.configurationFile` JVM parameter, I can see that `ApplicationContextProvider.getApplicationContext()` returns null.